LinkedList.js

```diff
@@ -0,0 +1,96 @@
+import { LinkedListElement } from "./LinkedListElement.js";
+
+export class LinkedList {
+  #head = null;
+  #tail = null;
+
+  addToTail(value) {
+    let newMember = null;
+    console.log("adding " + value);
+    if (this.#head === null) {
+      this.#head = new LinkedListElement(value, null, null);
+      this.#tail = this.#head;
+    } else if (this.#head.nextMember === null) {
+      newMember = new LinkedListElement(value, null, this.#head);
+      this.#head.nextMember = newMember;
+    } else {
+      newMember = new LinkedListElement(value, null, this.#tail);
+      this.#tail.nextMember = newMember;
+    }
+    this.#tail = newMember;
+  }
+
+  toArray() {
+    let currentMember = this.#head;
+    const array = [];
+    while (currentMember !== null) {
+      array.push(currentMember);
+      currentMember = currentMember.nextMember;
+    }
+    return array;
+  }
+
+  remove(elementNumber) {
+    const elementToRemove = this.getNthElement(elementNumber);
+    elementToRemove.nextMember.previousMember = elementToRemove.previousMember;
+    elementToRemove.previousMember.nextMember = elementToRemove.nextMember;
+  }
+
+  reverse = () => {
+    if (this.#tail && this.#head !== this.#tail) {
+      let reversedList = new LinkedList();
+      let currentMember = this.#tail;
+      while (currentMember !== null) {
+        reversedList.addToTail(currentMember.value);
+        currentMember = currentMember.previousMember;
+      }
+      return reversedList;
+    }
+  };
+  switchPositions(leftElementNumber, rightElementNumber) {
```

Comment:
as arguments pass a reference to the LinkedListElement, this way you don't need to call the `getNthElement` which is consuming N operations.

Reply:
done